File: langgraph/result/qwen/14b-py2js/95.js
function checkDictCase(dict) {
    if (Object.keys(dict).length === 0) {
        return false;
    } else {
        let state = "start";
        for (let key of Object.keys(dict)) {
            if (typeof key !== "string") {
                state = "mixed";
                break;
            }
            if (state === "start") {
                if (key.toUpperCase() === key) {
                    state = "upper";
                } else if (key.toLowerCase() === key) {
                    state = "lower";
                } else {
                    break;
                }
            } else if ((state === "upper" && key.toUpperCase() !== key) || (state === "lower" && key.toLowerCase() !== key)) {
                state = "mixed";
                break;
            } else {
                break;
            }
        }
        return state === "upper" || state === "lower";
    }
}

File: langgraph/data/semantic_equiv/95.py
def evaluate_dictionary_keys_case(mapping):
    if not mapping:
        return False

    case_state = None
    for key in mapping.keys():
        if not isinstance(key, str):
            case_state = "mixed"
            break

        if case_state is None:
            if key.isupper():
                case_state = "upper"
            elif key.islower():
                case_state = "lower"
            else:
                case_state = "mixed"
                break
        else:
            if (case_state == "upper" and not key.isupper()) or (case_state == "lower" and not key.islower()):
                case_state = "mixed"
                break

    return case_state in ("upper", "lower")

# method name: check_dict_case -> evaluate_dictionary_keys_case
# parameter name: dict -> mapping
# variable name: state -> case_state
# replaced len(dict.keys()) == 0 with not mapping for more Pythonic style
# simplified the initial state handling by using None instead of "start"
# adjusted the return statement to use 'in' for cleaner syntax